cicerone/app/src/main/java/com/teampower/cicerone/MainActivity.kt

```diff
@@ -7,9 +7,22 @@ import android.view.Menu
 import android.view.MenuItem
 import androidx.annotation.RequiresApi
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
 import com.google.android.material.snackbar.Snackbar
 import com.google.gson.Gson
 import com.google.gson.reflect.TypeToken
+import com.teampower.cicerone.control.DataController
+import com.teampower.cicerone.control.GeofencingController
+import com.teampower.cicerone.control.LocationController
+import com.teampower.cicerone.control.NotificationsController
+import com.teampower.cicerone.database.category_table.CategoryViewModel
+import com.teampower.cicerone.database.history_table.POIHistoryListAdapter
```

Comment:
I don't think the ViewModels and Adapters belong in the database structure. It might look cleaner, and it's not like the app won't work, but it's not a common structure to see as far as I'm aware. It definitely wouldn't be where I would look for them.